api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -106,6 +107,16 @@ message DiscoveryResponse {
   // [#not-implemented-hide:]
   // The control plane instance that sent the response.
   config.core.v3.ControlPlane control_plane = 6;
+
+  // [#not-implemented-hide:]
+  // The TTL feature is meant to support configurations that should be removed in the event of a management server
+  // failure. For example, the feature may be used for fault injection testing where the fault injection should be
+  // terminated in the event that Envoy loses contact with the management server.
+  //
+  // When a response is received with TTL set, a timer is started. Each subsequent response containing a TTL
+  // resets this timer to a new value. If a response is received with no TTL set, the timer is removed. Upon
+  // expiration of the timer, the configurations for all resources in the response are removed.
+  google.protobuf.Duration ttl = 7;
```

Comment:
If we wind up using the `Resource` proto in `DiscoveryResponse` as we're discussing elsewhere, we may not need to add this here.

Reply:
Yeah, I was discussing this on Slack, I'm still slightly in favor of this, depending on how quickly we can close on the xDS resource naming work.

Reply:
Yeah, I agree. I want to figure out if it's possible to send 'Resource' in the SotW DiscoveryResponse without breaking wire compatibility. New instances of envoy can introspect the 'resources' returned in the response and unwrap any Resources it finds. It will be necessary to avoid sending 'Resource' to older instances of envoy that don't have this wrapping logic. Do we send enough versioning information on the wire to make this determination?

Reply:
This sounds like something we could easily trigger via a client capability.